include/irr/asset/filters/CSwizzleAndConvertImageFilter.h

```diff
@@ -188,20 +188,50 @@ class CSwizzleAndConvertImageFilter<EF_UNKNOWN,EF_UNKNOWN,Swizzle> : public CIma
 			#endif
 			auto perOutputRegion = [&blockDims,inFormat,outFormat,&state](const CMatchedSizeInOutImageFilterCommon::CommonExecuteData& commonExecuteData, CBasicImageFilterCommon::clip_region_functor_t& clip) -> bool
 			{
-				auto swizzle = [&commonExecuteData,&blockDims,inFormat,outFormat,&state](uint32_t readBlockArrayOffset, core::vectorSIMDu32 readBlockPos)
+				const auto inTexelOrBlockByteSize = asset::getTexelOrBlockBytesize(inFormat);
+				const auto inChannelAmount = asset::getFormatChannelCount(inFormat);
+
+				// TESTING
+				auto getClampedSourcePixels = [&](auto sourcePixels)
+				{
+					constexpr auto MaxPlanes = 4;
+					auto buffer = core::make_smart_refctd_ptr<asset::ICPUBuffer>(blockDims.y * blockDims.x * inTexelOrBlockByteSize);
+					const void* newSourcePixels[MaxPlanes] = { buffer->getPointer(), nullptr, nullptr, nullptr };
+
+					for (auto blockY = 0u; blockY < blockDims.y; blockY++)
+						for (auto blockX = 0u; blockX < blockDims.x; blockX++)
+						{
+							std::array<double, MaxPlanes> decodeBuffer = {};
+							asset::decodePixelsRuntime(inFormat, sourcePixels, decodeBuffer.data(), blockX, blockY);
+
+							for (auto& value : decodeBuffer)
```

Comment:
I'd do a loop from i=0 to output format channel count, if the output format is not `EF_UNDEFINED` (not runtime specified)

Reply:
for this class a `for (auto i=0; i<MaxChannels; i++)` is desired, for the compile-time general case
> 
> 
> you've only implemented the clamping for runtime (or at least 1-format-runtime, because we dont bother specializing for half-compile-time cases) format conversion.
> 
> I need an implementation for the both-formats-specified-compile-time case, which is the general class above, where `swizzle` lambda is defined between line 144 and 160

I want the for from 0 to output format's channel count (given by the template function, not runtime function)

Reply:
adjusted for runtime in following commit, compile time left